test/jdk/tools/jpackage/helpers/jdk/jpackage/test/JPackageCommand.java

```diff
@@ -298,6 +299,57 @@ public JPackageCommand setFakeRuntime() {
         return this;
     }
 
+    public void createJPackageXMLFile(String mainLauncher, String mainClass)
+            throws IOException {
+        String appImage = getArgumentValue("--app-image");
+        if (appImage == null) {
+            throw new RuntimeException("Error: --app-image expected");
+        }
+
+        final String version = System.getProperty("java.version");
+        final String platform;
+
+        if (TKit.isWindows()) {
+            platform = "windows";
+        } else if (TKit.isLinux()) {
+            platform = "linux";
+        } else if (TKit.isOSX()) {
+            platform = "macOS";
+        } else {
+            platform = "unknown";
+        }
+
+        ApplicationLayout layout = ApplicationLayout.platformAppImage();
+        Path jpackageXMLFile = layout.resolveAt(Path.of(appImage))
+                .appDirectory().resolve(".jpackage.xml");
```

Comment:
Can be simplified:
`Path jpackageXMLFile = AppImageFile.getPathInAppImage(appImage);`